parquet/parquet.cpp

```diff
@@ -1356,7 +1356,7 @@ static void add_null(pq_column *c, int16_t r, int16_t d)
     ss << "column '" << c->n->name << "' is required";
     throw pq::ParquetException(ss.str());
   }
-  update_levels(c, r, d);
+  update_levels(c, r, d - 1 ? d > 0 : 0);
```

Comment:
Tricky stuff.

I saw the breakage you describe in `example.parquet`. While investigating this I discovered that lists were encoding nulls improperly as well. I don't know if there's a more general solution needed but the use of `maybe_dissect_field`in `dissect_list` and `dissect_map` satisfies the cases I tested.